sign/bls/bls.go

```diff
@@ -31,6 +31,29 @@ func Sign(suite pairing.Suite, x kyber.Scalar, msg []byte) ([]byte, error) {
 	return s, nil
 }
 
+// AggregateSignatures combines signatures created using the Sign function
+func AggregateSignatures(suite pairing.Suite, sigs ...[]byte) ([]byte, error) {
+	sig := suite.G1().Point()
+	for _, sigBytes := range sigs {
+		sigToAdd := suite.G1().Point()
+		if err := sigToAdd.UnmarshalBinary(sigBytes); err != nil {
+			return nil, err
+		}
+		sig.Add(sig, sigToAdd)
+	}
+	return sig.MarshalBinary()
+}
+
+// VerifyMultiSignature takes an aggregated signature and a slice of public Points
+// and verifies that the signature is valid
+func VerifyMultiSignature(suite pairing.Suite, Xs []kyber.Point, msg, sig []byte) error {
```

Comment:
Having played around with multisigs already, I find it better to expect the aggregated signature AND aggregated public key already. Usually, you select which public key and signatures to aggregate from a bitset for example. When iteratating the bitset, you can already aggregate, instead of creating a temporary slice containing only the public keys to aggregate to be able to give it to this function. Plus, it's also more consistent to either expect aggregated stuff or individual stuff separately.

Reply:
I got rid of this method, and added a AggregateKeys method